### DISCUSSION TITLE
Console markup in Text()

### DISCUSSION THREAD

**Discussion by ewels** [Textualize/rich#1951] [Q&A] [answered]
Sorry if I'm missing something. Is there a way to get console markup to work with `Text()` objects?

```python
>>> console.print(Text("[red]some[/] text"))

[red]some[/] text
```

**Comment by willmcgugan** [comment, marked as answer]
Yep, `Text.from_markup("[red]some[/] text")`

**Reply by ewels** [reply]
I had a feeling it was an easy answer that I'd missed somewhere! Thanks!

**Reply by ewels** [reply]
Is there a way to do this together with `mytext.append()`?

**Reply by willmcgugan** [reply]
Append will accept another Text object.

**Reply by ewels** [reply]
Ahh, smart!

**Reply by ewels** [reply]
ok ok last one, promise (hopefully). Is it possible to convert a `Markdown` object to `Text`? I'm trying to do something like this:

```python
# Handle text with rich markup or markdown
def make_rich_rext(text, style):
    if USE_MARKDOWN:
        text = Markdown(text)
    if USE_RICH_MARKUP:
        text = Text.from_markup(text, style=style)
    else:
        text = Text(text, style=style)
    return text
```

**Reply by willmcgugan** [reply]
Afraid not. But why do you need the conversion? You can `console.print` either a Markdown instance or a Text object.

Maybe you are generating a more complex output that can be built with Text. You could use [groups](https://rich.readthedocs.io/en/latest/group.html).

**Reply by ewels** [reply]
Perfect! The `@groups()` decorator works a treat, thanks 🚀 

https://github.com/ewels/rich-click/commit/6309974bfc9eda3424473cfaadf93cb73da7cb2e (bit of a messy commit, but using it to stick various chunks of text together into a panel, which can now be rich markup Text or Markdown...)

**Reply by ewels** [reply]
Ack, ok _final_ niggle. Groups works beautifully, but adds in a newline after each renderable.

So for example:

```python
from rich import print
from rich.console import Group
from rich.text import Text
from rich.panel import Panel

panel_group = Group(
    Text("Hello", style="on blue"),
    Text("World", style="on red"),
)
print(Panel(panel_group))
```

<img width="309" alt="image" src="https://user-images.githubusercontent.com/465550/154558167-035d9bf9-dfef-4a01-97af-c7a6fbfe872d.png">


Is there any way to get things in line with a Group? More like this:
```python
print(Text("Hello", style="on blue"), Text("World", style="on red"))
```

<img width="310" alt="image" src="https://user-images.githubusercontent.com/465550/154558224-9297c7fe-4aa4-4c39-a66a-4111ad68e875.png">

**Reply by willmcgugan** [reply]
You can use a table for your panel contents.. Table.grid() will create a borderless table. Another option is Columns, which will optimally fit a number of items.

**Reply by ewels** [reply]
Beauty! On point as ever, `Columns` works perfectly.. Thanks!

https://github.com/ewels/rich-click/commit/a004ad47f6d18923f2c32cce33c53ccc80e55e17
